src/graphql/mod.rs

```diff
@@ -97,21 +98,66 @@ impl QueryClient {
         Q::ResponseData: Debug,
     {
         let body = Q::build_query(variables);
-        let request = self.inner().post(&self.api_url).json(&body);
+        let client = self.inner();
 
         debug!("url: {:?}", &self.api_url);
-        debug!("request: {:#?}", request);
         debug!("GraphQL query: \n{}", body.query);
 
-        let response: Response<Q::ResponseData> = request.send().await?.json().await?;
+        let response: Result<Response<Q::ResponseData>, APIError> =
+            self.retry_request::<Q>(client, body, handler).await;
         debug!("GraphQL response: {:#?}", response);
 
-        if let Some(errors) = response.errors.clone() {
-            let first_error = errors[0].clone();
+        response
+    }
+
+    async fn retry_request<Q>(
+        &self,
+        client: &reqwest::Client,
```

Comment:
I think we don't have to pass `client` here, we can get the client value from `self`